Symfony 5.4 + PHP 8.1

ruudk opened the pull request:
Best to review commit by commit.

Symfony 6 will be added in another PR, as it has a blocker:
- https://github.com/malukenho/docheader/pull/35

ruudk commented:
@othillo Tests are not running. Can you setup branch protection so that it's visible that it requires approval?

othillo commented:
> @othillo Tests are not running. Can you setup branch protection so that it's visible that it requires approval?
Yeah we only run them for PRs to master. Feel free to cherry-pick my first commit and rebase to master.

ruudk commented:
~Why only on master? Every PR should be tested, right?~

Wow, didn't even realize it targetted it like this. Didn't choose to do that! Magic ✨

ruudk committed:
PHPStan > Rename `excludes_analyse` to `excludePaths`

Needed for PHPStan 1.0.

ruudk committed:
Use Flex to pin to Symfony versions

ruudk committed:
Normalize composer.json

ruudk committed:
Composer > Drop PHP 7.2

We don't test for this anymore.

ruudk committed:
Drop Symfony 3.4

We don't test for this anymore.

ruudk committed:
Bump minimum Symfony version from 5.0 to 5.3

It's still possible to install on Symfony 4.4

ruudk committed:
Remove BC layer for Symfony <= 4.1

ruudk committed:
Add type hints

ruudk committed:
Test on Symfony 5.4

ruudk committed:
Drop PHP 7.3

It's EOL.

ruudk committed:
Test on PHP 8.1

ruudk committed:
Do not fail fast on matrix

This makes testing multiple versions very annoying.

ruudk committed:
Test on Symfony 4.4.20

This is needed for php-cs-fixer that is a transitive dependency of broadway/coding-standard.

ruudk committed:
Use phpstan-symfony plugin

This way PHPStan understands the TreeBuilder.

ruudk commented:
@othillo Ready for review.

Symfony 6 will need a separate PR.

othillo commented on test/Command/CommandMetadataEnricherTest.php:
```diff
@@ -14,26 +14,52 @@
+    /**
+     * @var MyCommand
+     */
     private $command;
```
we might as well use typed properties here now that we are on PHP 7.4

ruudk replied:
I was temped to do so, but I don't want to change things that are not needed as the PR is already huge.

othillo replied:
Good call!

othillo commented:
Thank you very much, @ruudk !
